lib/reporter.js

```diff
@@ -218,58 +219,72 @@ var CoverageReporter = function (rootConfig, helper, logger, emitter) {
+      for (const reporterConfig of reporters) {
+        await Promise.all(browsers.map(async (browser) => {
+          const coverageMap = coverageMaps[browser.id]
+          if (!coverageMap) {
+            return
+          }
 
-        // If config.check is defined, check coverage levels for each browser
-        if (hasOwnProperty.call(config, 'check') && !checkedCoverage[browser.id]) {
-          checkedCoverage[browser.id] = true
-          var coverageFailed = checkCoverage(browser, remappedCoverageMap)
-          if (coverageFailed && results) {
-            results.exitCode = 1
+          const mainDir = reporterConfig.dir || config.dir
+          const subDir = reporterConfig.subdir || config.subdir
+          const outputPath = generateOutputPath(basePath, browser.name, mainDir, subDir)
+          const remappedCoverageMap = await sourceMapStore.transformCoverage(coverageMap)
+
+          const options = helper.merge(config, reporterConfig, {
+            dir: outputPath,
+            subdir: '',
+            browser: browser,
+            emitter: emitter,
+            coverageMap: remappedCoverageMap
+          })
+
+          // If config.check is defined, check coverage levels for each browser
+          if (hasOwnProperty.call(config, 'check') && !checkedCoverage[browser.id]) {
+            checkedCoverage[browser.id] = true
+            var coverageFailed = checkCoverage(browser, remappedCoverageMap)
+            if (coverageFailed) {
+              exitCode = 1
+            }
           }
-        }
 
-        const context = istanbulLibReport.createContext(options)
-        const report = reports.create(reporterConfig.type || 'html', options)
+          const context = istanbulLibReport.createContext(options)
+          const report = reports.create(reporterConfig.type || 'html', options)
 
-        // // If reporting to console or in-memory skip directory creation
-        const toDisk = !reporterConfig.type || !reporterConfig.type.match(/^(text|text-summary|in-memory)$/)
+          // // If reporting to console or in-memory skip directory creation
+          const toDisk = !reporterConfig.type || !reporterConfig.type.match(/^(text|text-summary|in-memory)$/)
 
-        if (!toDisk && reporterConfig.file === undefined) {
-          report.execute(context)
-          return
-        }
+          if (!toDisk && reporterConfig.file === undefined) {
+            report.execute(context)
+            return
+          }
 
-        helper.mkdirIfNotExists(outputPath, function () {
-          log.debug('Writing coverage to %s', outputPath)
-          report.execute(context)
-        })
-      }))
-    }
+          helper.mkdirIfNotExists(outputPath, function () {
+            log.debug('Writing coverage to %s', outputPath)
+            report.execute(context)
+          })
+        }))
+      }
+      return exitCode
+    })
   }
 
-  this.onExit = function (done) {
+  this.onExit = function (exitDone) {
+    const done = () => {
```

Comment:
Custom done can be execute with 1 or 0. I think, it is regression

Reply:
As of now, the `onExit` done callback from karma-runner doesn't expect any value. This is a new feature proposal from my PR (https://github.com/karma-runner/karma/pull/3541). So I don't understand how this can be a regression.

I don't know how the `_onExit` is used or how implements it. But I expected it to be a hook for some third-party and I intentionally did not allow it to provide an exit code value.